addons/pkg/webhooks/webhooks.go

```diff
@@ -32,14 +32,19 @@ type WebhookTLS struct {
 }
 
 const (
-	oneWeek = time.Hour * 24 * 7
-	oneDay  = time.Hour * 24
+	oneWeek               = time.Hour * 24 * 7
+	oneDay                = time.Hour * 24
+	RequireMinGracePeriod = time.Hour * 24
 )
 
 func (w *WebhookTLS) UpdateOrCreate() error {
-	if w.RotationTime > oneWeek {
-		w.Logger.Info("rotation time will be set to maximum allowed value of one Week")
-		w.RotationTime = oneWeek
+	// We need to have enough time (RequireMinGracePeriod) to rotate the certificates before their end of life (one week) to avoid race conditions.
+	// The function we are solving for is  "RotationTime + RequiredMinGracePeriod <= oneWeek && RotationTime > 0"
+	maxAllowedRotationTime := oneWeek - RequireMinGracePeriod // ensures gracePeriod > 0
+
+	if w.RotationTime > maxAllowedRotationTime {
+		w.Logger.Info("rotation time will be set to maximum allowed value of " + maxAllowedRotationTime.String())
+		w.RotationTime = maxAllowedRotationTime
 	}
```

Comment:
Looking at how to cover this lines in this patch (they are covered by other tests but don't register with codecov).

Reply:
There is no easy way to fix codecov because this function needs to make calls against a live cluster api server. 
The tests are part of this PR (https://github.com/vmware-tanzu/tanzu-framework/pull/4317/files#diff-ca0a0adf75f2bf96857857f00a1312c134e1525e2c6a98f84330ae097a51b5e6R111) but codecov is not registering them, probably because they are not in the same directory.